### ISSUE TITLE
Update help for Arguments

### ISSUE THREAD

**Issue by sayedihashimi** [dotnet/templating#129]
When I run `dotnet new3 --help`  for arguments I see the following.

```bash
Usage:  [arguments] [options]

Arguments:
  template  The template to instantiate.
```

I think it would be more consistent with `dotnet --help` if we output the following.

```bash
Usage:  [template] [options]

Arguments:
  [template]  Name of the template used for creation
  [options]   Options

Options:
  -l|--list         List templates containing the specified name.
  -lang|--language  Specifies the language of the template to create
  -n|--name         The name for the output being created. If no name is specified, the name of the current directory is used.
  -o|--output       Location to place the generated output.
  -h|--help         Displays help for this command.


```

**Comment by seancpeters** [comment]
Looking into this more, only "dotnet --help" displays help as shown above. Running "dotnet <command> --help" displays similarly to how new3 displays help. For example:

```
c:\GitHub\templating>dotnet new --help
.NET Initializer

Usage: dotnet new [options]

Options:
  -h|--help             Show help information
  -l|--lang <LANGUAGE>  Language of project    Valid values: C#, F#.
  -t|--type <TYPE>      Type of project        Valid values for C#: Console, Web, Lib, Mstest, Xunittest. Valid values for F#: Console, Web, Lib, Mstest, Xunittest.

c:\GitHub\templating>dotnet restore --help
restore

Usage: restore [arguments] [options] [[--] <arg>...]]

Arguments:
  [root]  Optional path to a project file or MSBuild arguments.

Options:
  -h|--help                       Show help information
  -s|--source <source>            Specifies a NuGet package source to use during the restore.
  --packages <packagesDirectory>  Directory to install packages in.
  --disable-parallel              Disables restoring multiple projects in parallel.
  --configfile <file>             The NuGet configuration file to use.
  --no-cache                      Do not cache packages and http requests.
  --ignore-failed-sources         Treat package source failures as warnings.
  --no-dependencies               Set this flag to ignore project to project references and only restore the root project
  -v|--verbosity                  Set the verbosity level of the command. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic]

Args:
  Any extra options that should be passed to MSBuild. See 'dotnet msbuild -h' for available options.
```

**Comment by mlorbetske** [comment]
This has been superseded by #725
